// @storybook/addon-ondevice-controls@6.5.7/dist/ControlsPanel.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
const react_native_theming_1 = require("@storybook/react-native-theming");
const react_1 = __importStar(require("react"));
const hooks_1 = require("./hooks");
const NoControlsWarning_1 = __importDefault(require("./NoControlsWarning"));
const PropForm_1 = __importDefault(require("./PropForm"));
const ButtonTouchable = react_native_theming_1.styled.TouchableOpacity(({ theme }) => ({
    backgroundColor: theme.button.secondary.backgroundColor,
    borderRadius: theme.button.secondary.borderRadius,
    borderWidth: theme.button.secondary.borderWidth,
    borderColor: theme.button.secondary.borderColor,
    paddingVertical: theme.button.paddingVertical,
    paddingHorizontal: theme.button.paddingHorizontal,
    justifyContent: 'center',
    alignItems: 'center',
}));
const ButtonText = react_native_theming_1.styled.Text(({ theme }) => ({
    color: theme.button.secondary.textColor,
    fontSize: theme.button.fontSize,
    fontWeight: theme.button.fontWeight,
}));
const ControlsPanel = ({ api }) => {
    const store = api.store();
    const storyId = store.getSelection()?.storyId;
    const [isPristine, setIsPristine] = (0, react_1.useState)(true);
    const [argsFromHook, updateArgs, resetArgs] = (0, hooks_1.useArgs)(storyId, store);
    const { argsObject, argTypes, parameters } = react_1.default.useMemo(() => {
        const { argTypes: storyArgTypes, parameters: storyParameters } = store.fromId(storyId);
        const storyArgsObject = Object.entries(storyArgTypes).reduce((prev, [key, argType]) => {
            const isControl = Boolean(argType?.control);
            return isControl
                ? {
                    ...prev,
                    [key]: {
                        ...argType,
                        name: key,
                        type: argType?.control?.type,
                        value: argsFromHook[key],
                    },
                }
                : prev;
        }, {});
        return {
            argTypes: storyArgTypes,
            parameters: storyParameters,
            argsObject: storyArgsObject,
        };
    }, [store, storyId, argsFromHook]);
    const hasControls = Object.keys(argTypes).length > 0;
    const isArgsStory = parameters.__isArgsStory;
    const showWarning = !(hasControls && isArgsStory);
    const updateArgsOnFieldChange = (0, react_1.useCallback)((args) => {
        updateArgs(args);
        setIsPristine(false);
    }, [updateArgs]);
    const handleReset = (0, react_1.useCallback)(() => {
        resetArgs();
        setIsPristine(true);
    }, [resetArgs]);
    if (showWarning) {
        return (0, jsx_runtime_1.jsx)(NoControlsWarning_1.default, {});
    }
    return ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(0, jsx_runtime_1.jsx)(PropForm_1.default, { args: argsObject, isPristine: isPristine, onFieldChange: updateArgsOnFieldChange }), (0, jsx_runtime_1.jsx)(ButtonTouchable, { onPress: handleReset, children: (0, jsx_runtime_1.jsx)(ButtonText, { children: "RESET" }) })] }));
};
exports.default = ControlsPanel;
