// @storybook/addon-ondevice-controls@6.5.7/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.register = exports.PARAM_KEY = exports.ADDON_ID = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const addons_1 = __importStar(require("@storybook/addons"));
const ControlsPanel_1 = __importDefault(require("./ControlsPanel"));
const Panel_1 = require("./Panel");
exports.ADDON_ID = 'RNCONTROLS';
exports.PARAM_KEY = 'controls';
function register() {
    addons_1.default.register(exports.ADDON_ID, (api) => {
        addons_1.default.addPanel(exports.ADDON_ID, {
            type: addons_1.types.PANEL,
            title: 'Controls',
            render: ({ active, key }) => ((0, jsx_runtime_1.jsx)(Panel_1.AddonPanel, { active: active, children: (0, jsx_runtime_1.jsx)(ControlsPanel_1.default, { api: api }) }, key)),
            paramKey: exports.PARAM_KEY,
        });
    });
}
exports.register = register;
