// @bacons/mdx@0.3.8/build/headings.jsx
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.H6 = exports.H5 = exports.H4 = exports.H3 = exports.H2 = exports.H1 = exports.em = void 0;
const react_1 = __importStar(require("react"));
const react_native_1 = require("react-native");
function em(value) {
    if (react_native_1.Platform.OS === "web")
        return `${value}em`;
    return react_native_1.PixelRatio.getFontScale() * 16 * value;
}
exports.em = em;
function createHeadingComponent(level) {
    const nativeProps = react_native_1.Platform.select({
        web: {
            "aria-level": level,
            role: "header",
        },
        default: {
            accessibilityRole: "header",
        },
    });
    return (0, react_1.forwardRef)((props, ref) => {
        return (<react_native_1.Text {...nativeProps} {...props} style={[styles[`h${level}`], props.style]} ref={ref}/>);
    });
}
exports.H1 = createHeadingComponent(1);
exports.H2 = createHeadingComponent(2);
exports.H3 = createHeadingComponent(3);
exports.H4 = createHeadingComponent(4);
exports.H5 = createHeadingComponent(5);
exports.H6 = createHeadingComponent(6);
if (__DEV__) {
    exports.H1.displayName = "H1";
    exports.H2.displayName = "H2";
    exports.H3.displayName = "H3";
    exports.H4.displayName = "H4";
    exports.H5.displayName = "H5";
    exports.H6.displayName = "H6";
}
// Default web styles: http://trac.webkit.org/browser/trunk/Source/WebCore/css/html.css
const styles = react_native_1.StyleSheet.create({
    h1: {
        // @ts-ignore
        fontSize: em(2),
        fontWeight: "bold",
    },
    h2: {
        // @ts-ignore
        fontSize: em(1.5),
        fontWeight: "bold",
    },
    h3: {
        // @ts-ignore
        fontSize: em(1.17),
        fontWeight: "bold",
    },
    h4: {
        // @ts-ignore
        fontSize: em(1),
        fontWeight: "bold",
    },
    h5: {
        // @ts-ignore
        fontSize: em(0.83),
        fontWeight: "bold",
    },
    h6: {
        // @ts-ignore
        fontSize: em(0.67),
        fontWeight: "bold",
    },
});
